umm-lib/src/cmr/umm/iso_smap/granule/orbit.clj

```diff
@@ -0,0 +1,111 @@
+(ns cmr.umm.iso-smap.granule.orbit
+  (:require
+   [clojure.data.xml :as x]
+   [clojure.string :as str]
+   [cmr.common.log :refer [info debug error]]
+   [cmr.common.util :as util]
+   [cmr.common.validations.core :as v]
+   [cmr.common.xml :as cx]
+   [cmr.spatial.encoding.gmd :as gmd]
+   [cmr.spatial.messages :as msg]
+   [cmr.spatial.validation :as sv]
+   [cmr.umm.umm-granule :as g])
+  (:import cmr.umm.umm_granule.Orbit))
+
+(defn- start-end-direction
+  [field-path direction]
+  (when (and (not= :asc direction)
+             (not= :desc direction))
+    {field-path [(msg/start-end-direction direction)]}))
+
+(defn- convert-direction
+  "Converts :asc <=> A and :desc <=> D"
+  [direction]
+  (cond
+    (= "A" direction) :asc
+    (= "D" direction) :desc
+    (= :asc direction) "A"
+    (= :desc direction) "D"
+    :default nil))
+
+(def validations
+  [{:ascending-crossing [v/required v/number (v/within-range -180.0 180.0)]
+    :start-lat [v/required v/number (v/within-range -90.0 90.0)]
+    :end-lat [v/required v/number (v/within-range -90.0 90.0)]
+    :start-direction start-end-direction
+    :end-direction start-end-direction}])
+
+(extend-protocol sv/SpatialValidation
+  cmr.umm.umm_granule.Orbit
+  (validate
+    [record]
+    (v/create-error-messages (v/validate validations record))))
+
+(defn- build-orbit-string
+  "Builds ISO SMAP orbit string using umm values."
+  [orbit]
+  (let [{:keys [ascending-crossing start-lat start-direction end-lat end-direction]} orbit]
+    (format "AscendingCrossing: %s StartLatitude: %s StartDirection: %s EndLatitude: %s EndDirection: %s"
+            ascending-crossing
+            start-lat (convert-direction start-direction)
+            end-lat (convert-direction end-direction))))
+
+(defn- coerce-float
+  "Coerce's string to float, catches exceptions and produces error message."
+  [field value]
+  (try
+    (Float. value)
+    (catch Exception e
+      (error e (format "For orbit field [%s] the value [%s] is not a number." field value)))))
```

Comment:
We're going to return whatever the log message error macro returns - right now that's `nil`. I think generally we'd want to add an explicit return - so add `nil` after logging the error if that's the value we want it to return.